### ISSUE TITLE
Gradient explosion at 17 epochs

### ISSUE THREAD

**Issue by NewEricWang** [JusperLee/Dual-Path-RNN-Pytorch#13]
Loss becomes bigger at epoch 17 when training the mode on our own dataset.

20-07-15 02:34:45 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:16000, lr:5.000e-04, loss:-12.715>
20-07-15 02:50:07 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:17000, lr:5.000e-04, loss:-12.711>
20-07-15 03:05:29 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:18000, lr:5.000e-04, loss:-12.701>
20-07-15 03:20:51 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:19000, lr:5.000e-04, loss:-12.699>
20-07-15 03:36:12 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:20000, lr:5.000e-04, loss:-12.700>
20-07-15 03:51:34 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:21000, lr:5.000e-04, loss:-12.703>
20-07-15 04:06:56 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:22000, lr:5.000e-04, loss:-12.704>
20-07-15 04:22:18 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:23000, lr:5.000e-04, loss:-12.703>
20-07-15 04:37:40 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:24000, lr:5.000e-04, loss:-12.705>
20-07-15 04:53:01 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:25000, lr:5.000e-04, loss:-12.703>
20-07-15 05:08:23 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:26000, lr:5.000e-04, loss:-12.705>
20-07-15 05:23:45 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:27000, lr:5.000e-04, loss:-12.703>
20-07-15 05:39:06 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:28000, lr:5.000e-04, loss:-12.705>
20-07-15 05:54:28 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:29000, lr:5.000e-04, loss:-12.704>
20-07-15 06:09:51 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:30000, lr:5.000e-04, loss:-12.704>
20-07-15 06:25:12 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:31000, lr:5.000e-04, loss:-12.705>
20-07-15 06:40:34 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:32000, lr:5.000e-04, loss:-12.705>
20-07-15 06:55:56 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:33000, lr:5.000e-04, loss:-12.705>
20-07-15 07:11:18 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:34000, lr:5.000e-04, loss:-12.704>
20-07-15 07:26:42 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:35000, lr:5.000e-04, loss:-12.702>
20-07-15 07:42:04 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:36000, lr:5.000e-04, loss:-12.698>
20-07-15 07:57:26 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:37000, lr:5.000e-04, loss:-12.599>
20-07-15 08:12:52 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:38000, lr:5.000e-04, loss:-8.057>
20-07-15 08:28:16 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:39000, lr:5.000e-04, loss:-3.748>
20-07-15 08:43:40 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:40000, lr:5.000e-04, loss:0.346>
20-07-15 08:59:05 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:41000, lr:5.000e-04, loss:4.240>
20-07-15 09:14:31 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:17, iter:42000, lr:5.000e-04, loss:7.948>
20-07-15 09:28:02 [/trainer/trainer_Dual_RNN.py:106 - INFO ] Finished *** <epoch:17, iter:42879, lr:5.000e-04, loss:11.061, Total time:659.562 min>
20-07-15 09:28:02 [/trainer/trainer_Dual_RNN.py:111 - INFO ] Start Validation from epoch: 17, iter: 0
20-07-15 09:34:36 [/trainer/trainer_Dual_RNN.py:136 - INFO ] <epoch:17, iter:1000, lr:5.000e-04, loss:160.000>
20-07-15 09:41:11 [/trainer/trainer_Dual_RNN.py:136 - INFO ] <epoch:17, iter:2000, lr:5.000e-04, loss:160.000>
20-07-15 09:42:57 [/trainer/trainer_Dual_RNN.py:142 - INFO ] Finished *** <epoch:17, iter:2267, lr:5.000e-04, loss:159.929, Total time:14.915 min>
20-07-15 09:42:57 [/trainer/trainer_Dual_RNN.py:171 - INFO ] No improvement, Best Loss: -12.5711
20-07-15 09:42:57 [/trainer/trainer_Dual_RNN.py:70 - INFO ] Start training from epoch: 18, iter: 0
20-07-15 09:58:27 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:1000, lr:5.000e-04, loss:160.000>
20-07-15 10:14:02 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:2000, lr:5.000e-04, loss:160.000>
20-07-15 10:29:38 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:3000, lr:5.000e-04, loss:160.000>
20-07-15 10:45:08 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:4000, lr:5.000e-04, loss:160.000>
20-07-15 11:00:39 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:5000, lr:5.000e-04, loss:160.000>
20-07-15 11:16:10 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:6000, lr:5.000e-04, loss:160.000>
20-07-15 11:31:38 [/trainer/trainer_Dual_RNN.py:100 - INFO ] <epoch:18, iter:7000, lr:5.000e-04, loss:160.000>

**Comment by JusperLee** [comment]
Yes, there will be this phenomenon, you need to reduce the learning rate by half.

**Comment by NewEricWang** [comment]
@JusperLee Thanks a lot!
